docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,953 @@
+
+```json
+{
+  "rpc": "http://127.0.0.1:9656/ext/bc/2jDWMrF9yKK8gZfJaaaSfACKeMasiNgHmuZip5mWxUfhKaYoEU/rpc",
+  "chainId": 99999
+}
+```
+
+#### Step 7.5: Run Tests
+
+Going to `./contract-examples`, we can finally run our tests by running the command below.
+
+```bash
+npx hardhat test --network local
+```
+
+Great they passed! All the functions implemented in the precompile work as expected!
+
+If your tests failed, please retrace your steps. Most likely the error is that the precompile was not enabled and some code is missing. Please also use the [official tutorial](https://github.com/ava-labs/hello-world-official-precompile-tutorial/pull/1) to double check your work as well.
+
+### Step 8: Create Genesis
+
+We can move our genesis file we created in the last step to `./tests/e2e/genesis/hello_world.json`.
+
+```bash
+cp /tmp/subnet-evm-genesis.json $GOPATH/src/github.com/ava-labs/subnet-evm/tests/e2e/genesis/hello_world.json
+```
+
+### Step 9: Add E2E tests
+
+In `./tests/e2e/solidity/suites.go` we can now write our first e2e test! The below code snippet can be copied and pasted to add the new test.
+
+```go
+ginkgo.It("hello world", func() {
+		err := startSubnet("./tests/e2e/genesis/hello_world.json")
+		gomega.Expect(err).Should(gomega.BeNil())
+		running := runner.IsRunnerUp()
+		gomega.Expect(running).Should(gomega.BeTrue())
+		runHardhatTests("./test/ExampleHelloWorld.ts")
+		stopSubnet()
+		running = runner.IsRunnerUp()
+		gomega.Expect(running).Should(gomega.BeFalse())
+	})
+
+	// ADD YOUR PRECOMPILE HERE
+	/*
+			ginkgo.It("your precompile", func() {
+			err := startSubnet("./tests/e2e/genesis/{your_precompile}.json")
+			gomega.Expect(err).Should(gomega.BeNil())
+			running := runner.IsRunnerUp()
+			gomega.Expect(running).Should(gomega.BeTrue())
+			runHardhatTests("./test/Example{YourPrecompile}Test.ts")
+			stopSubnet()
+			running = runner.IsRunnerUp()
+			gomega.Expect(running).Should(gomega.BeFalse())
+		})
+	*/
+```
+
+### Step 10: Run E2E Test
+
+We also need to modify the genesis in `./scripts/run.sh` to enable our precompile.
+Copy and paste the code snippet below and add it to to the genesis located in `./scripts/run.sh`.
+
+```json
+"helloWorldConfig": {
+  "blockTimestamp": 0
+},
+```
+
+Now we can run our script with some flags specified.
+
+```bash
+cd $GOPATH/src/github.com/ava-labs/subnet-evm
+SKIP_NETWORK_RUNNER_START=true SKIP_NETWORK_RUNNER_SHUTDOWN=true ENABLE_SOLIDITY_TESTS=true scripts/run.sh
+```
+
+### Conclusion
+
+We have now created a stateful precompile from scratch with the precompile generation tool. We hope you had fun and learned a little more about the Subnet-EVM. Now that you have created a simple stateful precompile, we urge you to create one of your own. If you have an idea for a stateful precompile that may be useful to the community, feel free to create a fork of the Subnet-EVM and create a pull request.
```

Comment:
```suggestion
We have now created a stateful precompile from scratch with the precompile generation tool. We hope you had fun and learned a little more about the Subnet-EVM. Now that you have created a simple stateful precompile, we urge you to create one of your own. If you have an idea for a stateful precompile that may be useful to the community, feel free to create a pull request on the [Subnet-EVM project](https://github.com/ava-labs/subnet-evm).
```

Reply:
thanks

Reply:
I will still use `fork` as we don't want them to make a PR on our version of the project.